src/es/leermanga/src/eu/kanade/tachiyomi/extension/es/leermanga/LeerManga.kt

```diff
@@ -0,0 +1,118 @@
+package eu.kanade.tachiyomi.extension.es.leermanga
+
+import eu.kanade.tachiyomi.multisrc.madara.Madara
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+
+class LeerManga : Madara(
+    "LeerManga",
+    "https://leermanga.net",
+    "es",
+) {
+    override val client: OkHttpClient = super.client.newBuilder()
+        .rateLimit(3)
+        .build()
+
+    override val supportsLatest = false
+
+    override val mangaSubString = "biblioteca"
+
+    // ============================== Popular ===============================
+
+    override fun popularMangaRequest(page: Int): Request =
+        GET("$baseUrl/$mangaSubString?page=$page", headers)
+
+    override val popularMangaUrlSelector = ".page-item-detail a"
+
+    override fun popularMangaNextPageSelector() = ".pagination li a[rel=next]"
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        if (genresList.isEmpty()) {
+            val document = Jsoup.parse(response.peekBody(Long.MAX_VALUE).string())
+            genresList = parseGenres(document)
+        }
+        return super.popularMangaParse(response)
+    }
+
+    override val filterNonMangaItems = false
+
+    // =============================== Search ===============================
+
+    override fun searchRequest(page: Int, query: String, filters: FilterList): Request {
+        var url = "$baseUrl/$mangaSubString".toHttpUrl().newBuilder()
+
+        if (query.isNotBlank()) {
+            url.addQueryParameter("search", query)
+        }
+
+        filters.forEach { filter ->
+            when (filter) {
+                is GenreGroup -> {
+                    val selected = filter.selected()
+                    if (selected.isBlank() || query.isNotBlank()) {
+                        return@forEach
+                    }
+                    url = selected.toHttpUrl().newBuilder()
+                }
+                else -> {}
+            }
+        }
+
+        url.addQueryParameter("page", "$page")
```

Comment:
Convention is the following, FWIW. Changing it is not required.
```suggestion
        url.addQueryParameter("page", page.toString())
```